src/redis.ts

```diff
@@ -15,31 +15,57 @@
  */
 
 import { Redis, RedisOptions } from 'ioredis';
+import { parseURL } from 'ioredis/built/utils';
 
 import { REDIS_CA_CERT, REDIS_CACHE_CA_CERT, REDIS_CACHE_URL, REDIS_URL } from './config.js';
 
-export function createClient(opts?: Partial<RedisOptions>): Redis {
-  const client = new Redis(REDIS_URL, {
-    tls:
-      REDIS_URL.startsWith('rediss') && REDIS_CA_CERT
-        ? {
-            ca: Buffer.from(REDIS_CA_CERT)
-          }
-        : undefined,
-    ...opts
-  });
-  return client;
+export const DEFAULT_REDIS_CONNECTION_OPTIONS: RedisOptions = {
+  ...parseURL(REDIS_URL),
+  tls:
+    REDIS_URL.startsWith('rediss') && REDIS_CA_CERT ? { ca: Buffer.from(REDIS_CA_CERT) } : undefined
+};
+
+export const DEFAULT_REDIS_CACHE_CONNECTION_OPTIONS: RedisOptions = {
+  ...parseURL(REDIS_CACHE_URL),
+  tls:
+    REDIS_CACHE_URL.startsWith('rediss') && REDIS_CACHE_CA_CERT
+      ? { ca: Buffer.from(REDIS_CACHE_CA_CERT) }
+      : undefined,
+  connectTimeout: 1000,
+  maxRetriesPerRequest: 1
+};
+
+export const sharedRedisClient = new Redis(DEFAULT_REDIS_CONNECTION_OPTIONS);
+export const sharedRedisCacheClient = new Redis(DEFAULT_REDIS_CACHE_CONNECTION_OPTIONS);
+
+const CLIENTS: Redis[] = [sharedRedisClient, sharedRedisCacheClient];
+
+export async function withRedisClient<R>(
```

Comment:
I added these `with` functions similar to python `with` context manager to make sure that the client is always closed.

I think it may be a useful pattern to use all around the codebase